From 7963ac18edd0bab3c961d6c77b6721011d0bdf08 Mon Sep 17 00:00:00 2001
From: Sreedhar Govardhana <srgovard@amd.com>
Date: Tue, 27 Jul 2021 12:44:33 -0500
Subject: [PATCH] ARM:dts:aspeed: Add device tree entries to enable KVM
 functionality

Add device tree entries for video and virtual USB hub

Signed-off-by: Sreedhar Govardhana <srgovard@amd.com>
---
 arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts | 22 ++++++++++++++++++++++
 arch/arm/boot/dts/aspeed-g6.dtsi          | 10 ++++++++++
 2 files changed, 32 insertions(+)

diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts b/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
index 7da02ea3e513..413f01b42314 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
@@ -26,6 +26,19 @@ memory@80000000 {
 		reg = <0x80000000 0x80000000>;
 	};

+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+
+		video_engine_memory: jpegbuffer {
+			size = <0x02000000>;	/* 32M */
+			alignment = <0x01000000>;
+			compatible = "shared-dma-pool";
+			reusable;
+		};
+	};
+
 };

 &mdio0 {
@@ -382,3 +395,12 @@ &jtag1 {
 	status = "okay";
 };

+&video {
+	status = "okay";
+	memory-region = <&video_engine_memory>;
+};
+
+&vhub {
+	status = "okay";
+};
+
diff --git a/arch/arm/boot/dts/aspeed-g6.dtsi b/arch/arm/boot/dts/aspeed-g6.dtsi
index 0300be29e46d..cac234073530 100644
--- a/arch/arm/boot/dts/aspeed-g6.dtsi
+++ b/arch/arm/boot/dts/aspeed-g6.dtsi
@@ -364,6 +364,16 @@ xdma: xdma@1e6e7000 {
 				status = "disabled";
 			};

+			video: video@1e700000 {
+				compatible = "aspeed,ast2600-video-engine";
+				reg = <0x1e700000 0x1000>;
+				clocks = <&syscon ASPEED_CLK_GATE_VCLK>,
+					 <&syscon ASPEED_CLK_GATE_ECLK>;
+				clock-names = "vclk", "eclk";
+				interrupts = <GIC_SPI 7 IRQ_TYPE_LEVEL_HIGH>;
+				status = "disabled";
+			};
+
 			gpio0: gpio@1e780000 {
 				#gpio-cells = <2>;
 				gpio-controller;
--
2.25.1

